src/formatDate.js

```diff
@@ -50,7 +50,51 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArr = date.split(fromFormat[3]);
+  const dateObj = {};
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    if (fromFormat[i].charAt(fromFormat[i].length - 1) === 'Y') {
+      dateObj.year = dateArr[i];
+    }
+
+    if (fromFormat[i].charAt(fromFormat[i].length - 1) === 'M') {
+      dateObj.month = dateArr[i];
+    }
+
+    if (fromFormat[i].charAt(fromFormat[i].length - 1) === 'D') {
+      dateObj.day = dateArr[i];
+    }
+  }
```

Comment:
instead of switch case for optimisation you can use `switch` case here instead of if statements